### PR TITLE
some fixes to return links

### PR DESCRIPTION
**Pull request by mikalauskas**
so i was away and didn't have much time to test #1003
this one work perfectly

### CODE REVIEW

File: app/views/agents/_table.html.erb [hunk 1]
```diff
@@ -53,7 +53,7 @@
         </td>
         <td class='<%= "agent-unavailable" if agent.unavailable? %>'>
           <% if agent.can_create_events? %>
-            <%= link_to(agent.events_count || 0, agent_events_path(agent)) %>
+            <%= link_to(agent.events_count || 0, agent_events_path(agent, return: (defined?(return_to) && return_to) || request.path)) %>
```
Comment (cantino) [inline app/views/agents/_table.html.erb]: Does this work better with `path` or `fullpath`?

File: app/views/events/index.html.erb [hunk 2]
```diff
@@ -40,7 +40,7 @@
 
       <% if @agent %>
         <div class="btn-group">
-          <%= link_to icon_tag('glyphicon-eye-open') + ' View Agent'.html_safe, agent_path(@agent, return: request.fullpath), class: "btn btn-default" %>
+          <%= link_to icon_tag('glyphicon-chevron-left') + ' Back'.html_safe, filtered_agent_return_link || agents_path, class: "btn btn-default" %>
```
Comment (cantino) [inline app/views/events/index.html.erb]: What was wrong with fullpath here?

**Comment by cantino** [comment]
Thanks @mikalauskas!

File: app/views/agents/_table.html.erb (hunk 1 above)
Comment (mikalauskas) [inline app/views/agents/_table.html.erb]: With fullpath it always returns you to Details page even if you came from scenario page

File: app/views/events/index.html.erb (hunk 2 above)
Comment (mikalauskas) [inline app/views/events/index.html.erb]: same as above

**Comment by cantino** [comment]
Thanks @mikalauskas!

**Pull request merged**